modules/core/src/main/java/org/apache/ignite/internal/processors/query/QueryTypeDescriptorImpl.java

```diff
@@ -646,11 +646,15 @@ else if (coCtx.kernalContext().cacheObjects().typeId(prop.type().getName()) !=
             if (propVal == null || prop.precision() == -1)
                 continue;
 
-            if (String.class == propVal.getClass() &&
-                ((String)propVal).length() > prop.precision()) {
-                throw new IgniteSQLException("Value for a column '" + prop.name() + "' is too long. " + 
-                    "Maximum length: " + prop.precision() + ", actual length: " + ((CharSequence)propVal).length(),
-                    isKey ? TOO_LONG_KEY : TOO_LONG_VALUE);
+            if (String.class == propVal.getClass() || byte[].class == propVal.getClass()) {
+                int propValLen = propVal.getClass() == String.class ? ((String) propVal).length()
```

Comment:
According to codestyle it should be
`((String)propVal).length())`
The same is for casting to `byte[]`